sale_general_term/models/sale_general_term.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class SaleGeneralTerm(models.Model):
+
+    _name = 'sale.general.term'
+    _description = 'Sale General Term'
+
+    @api.constrains('is_enabled', 'country_ids', 'lang')
+    def _check_is_enabled_country_ids(self):
+        for term in self:
+            enabled_terms = self.search([
```

Comment:
Maybe do the search outside of the loop and filter inside to avoid too much queries.

Reply:
@dreispt